cpp/tests/community/k_truss_test.cpp

```diff
@@ -0,0 +1,338 @@
+    auto [k_truss_usecase, input_usecase] = param;
+    raft::handle_t handle{};
+
+    HighResTimer hr_timer{};
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.start("SG Construct graph");
+    }
+
+    // NX k_truss is not implemented for graph with self loop and multi edges therefore dropped
+    // them especially for rmat generated graphs.
+    auto [graph, edge_weight, d_renumber_map_labels] =
+      cugraph::test::construct_graph<vertex_t, edge_t, weight_t, false, false>(
+        handle, input_usecase, k_truss_usecase.test_weighted_, renumber, true, true);
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.stop();
+      hr_timer.display_and_clear(std::cout);
+    }
+
+    auto graph_view = graph.view();
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.start("K-truss");
+    }
+
+    auto [d_cugraph_src, d_cugraph_dst, d_cugraph_wgt] =
+      cugraph::k_truss<vertex_t, edge_t, weight_t, false>(
+        handle,
+        graph_view,
+        edge_weight ? std::make_optional((*edge_weight).view()) : std::nullopt,
+        k_truss_usecase.k_,
+        false);
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.stop();
+      hr_timer.display_and_clear(std::cout);
+    }
+
+    if (k_truss_usecase.check_correctness_) {
+      std::optional<cugraph::graph_t<vertex_t, edge_t, false, false>> modified_graph{std::nullopt};
+
+      std::optional<
+        cugraph::edge_property_t<cugraph::graph_view_t<vertex_t, edge_t, false, false>, weight_t>>
+        modified_edge_weight{std::nullopt};
+      std::tie(*modified_graph, modified_edge_weight, std::ignore, std::ignore, std::ignore) =
+        cugraph::
+          create_graph_from_edgelist<vertex_t, edge_t, weight_t, edge_t, int32_t, false, false>(
+            handle,
+            std::nullopt,
+            std::move(d_cugraph_src),
+            std::move(d_cugraph_dst),
+            std::move(d_cugraph_wgt),
+            std::nullopt,
+            std::nullopt,
+            cugraph::graph_properties_t{true, false},
+            renumber);
+
+      // Convert cugraph results to CSR
+      auto [h_cugraph_offsets, h_cugraph_indices, h_cugraph_values] =
+        cugraph::test::graph_to_host_csr(
+          handle,
+          (*modified_graph).view(),
+          modified_edge_weight ? std::make_optional((*modified_edge_weight).view()) : std::nullopt,
+          std::optional<raft::device_span<vertex_t const>>(std::nullopt));
+
+      // Remove isolated vertices.
+      h_cugraph_offsets.erase(std::unique(h_cugraph_offsets.begin() + 1, h_cugraph_offsets.end()),
+                              h_cugraph_offsets.end());  // CSR start from 0
```

Comment:
Should we really do this? Can we just compare cugraph COO output and reference COO output?

Reply:
Ya but that means I will convert the reference ktruss to COO.

Reply:
I think I will have to rely on csr comparison because my cugraph results are not sorted. I get the same concern I had last week regarding sorting edges.

Reply:
Never mind. I extended the sorting utility functions in `thrust_wrapper` like you suggested.